.pylintrc

```diff
@@ -189,6 +189,9 @@ init-import=no
 # not used).
 dummy-variables-rgx=(unused)?_.*|dummy
 
+# A regular expression matching ignored argument variable names
+ignored-argument-names=(unused)?_.*|dummy
```

Comment:
What happened here? This PR adds ~50 `disable=unused-argument` lines. Can we not continue to have parameters named `unused_` automatically understood as known unused arguments?

Reply:
Seems that the regex in pylintrc is taken into account for `unused-variable` only, not `unused-argument` here.

However, using `_argument` instead of `unused_argument` effectively avoid pylint to complain about it. So I can also do that instead of adding `disable=unused-argument`.

As a general note, but it is not a part of this PR: I do not like this kind of decorator usage for mock here, as it ends up with weird unused injected arguments. A `with` context manager approach would be way better here.

Reply:
That's disappointing. Previously we were able to name variables `unused_` to make pylint happy which is a convention that has been heavily used in the code.

What I'd personally like to see in order of preference is:

1. Verify that there's no way to modify the `pylint` config to keep the current behavior we have where `unused_` variables are treated the same as `_` prefixed variables. If there's a way to configure `pylint` to do this, I think we should do it and remove all added `pylint: disable=` lines about this.
2. If 1 isn't possible, change all parameters whose name is `unused_` and the function isn't part of a public API (there probably isn't any overlap here but you never know...) to use a `_` prefix instead. If some of our `.pylintrc` for setting up `unused_` is no longer necessary, let's remove it.

Reply:
Have you specific reasons to not consider the `with` approach?

Reply:
I think making the changes you described is fine in the cases the unused parameter is a mock object.

Reply:
Finally, it appears that the relevant option was here: `ignored-argument-names`. It was just at the wrong place for recent versions of pylint. Now the arguments of type `unused_...` are effectively ignored by pylint.

I removed most of the `#pylint: disable=unused-argument` directives.